File: Selenium/Activities/Activity14.java
package example;

import java.util.List;

import org.openqa.selenium.By;
import org.openqa.selenium.WebDriver;
import org.openqa.selenium.WebElement;
import org.openqa.selenium.chrome.ChromeDriver;

public class Activity14 {

	public static void main(String[] args) {
		WebDriver driver = new ChromeDriver();
		driver.get("https://training-support.net/webelements/tables");
		System.out.println("Title of the page is: " + driver.getTitle());
		
		List<WebElement> row = driver.findElements(By.xpath("//table[contains(@class, 'w-full')]/tbody/tr"));
		System.out.println("Number Of Row's :- " +row.size());
		
		List<WebElement> col = driver.findElements(By.xpath("//table[contains(@class, 'w-full')]/thead/tr/th"));
		System.out.println("Number Of Col :- " +col.size());
		
		WebElement BookName = driver.findElement(By.xpath("//table[contains(@class, 'w-full')]/tbody/tr[5]/td[2]"));
		System.out.println("BookName :- " +BookName.getText());
		
		driver.findElement(By.xpath("//table[contains(@class, 'w-full')]/thead/tr/th[5]")).click();
		BookName = driver.findElement(By.xpath("//table[contains(@class, 'w-full')]/tbody/tr[5]/td[2]"));
		
		System.out.println("BookName After Sorting :- " +BookName.getText());
		driver.quit();
		
	}

}

File: Selenium/Activities/Activity14.py
from selenium import webdriver
from webdriver_manager.firefox import GeckoDriverManager
from selenium.webdriver.firefox.service import Service as fireFoxService
from selenium.webdriver.common.by import By
import time

service=fireFoxService(GeckoDriverManager().install())

with webdriver.Firefox(service=service)as driver:
    driver.get("https://www.training-support.net/selenium/tables")
    print("Page Title:- " + driver.title)

    row=driver.find_elements(By.XPATH,"//table[@id='sortableTable']//tbody//tr")
    col=driver.find_elements(By.XPATH,"//table[@id='sortableTable']//tbody//tr//td")
    print("Row Count:- " + str(len(row)))
    print("Col Count :- " + str(len(col)))

    secondCell=driver.find_element(By.XPATH,"//table[@id='sortableTable']//tbody//tr[2]//td[2]")
    print("Second Cell value :- " + secondCell.text)
    time.sleep(2)
    driver.find_element(By.XPATH,"//table[@id='sortableTable']//thead//th[text()='Name']").click()
    print("Header is clicked")

    secondCell=driver.find_element(By.XPATH,"//table[@id='sortableTable']//tbody//tr[2]//td[2]")
    print("Second Cell value :- " + secondCell.text)

    footer=driver.find_elements(By.XPATH,"//table[@id='sortableTable']//tfoot//tr//th")

    for value in footer:
        print(value.text)




